src/main/java/org/folio/innreach/batch/contribution/service/OngoingContributionStatusServiceImpl.java

```diff
@@ -0,0 +1,38 @@
+package org.folio.innreach.batch.contribution.service;
+
+import lombok.AllArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+import org.folio.innreach.domain.entity.ContributionStatus;
+import org.folio.innreach.domain.entity.OngoingContributionStatus;
+import org.folio.innreach.repository.OngoingContributionStatusRepository;
+import org.springframework.stereotype.Service;
+
+import static org.folio.innreach.domain.entity.ContributionStatus.RETRY;
+
+@Service
+@AllArgsConstructor
+@Log4j2
+public class OngoingContributionStatusServiceImpl implements OngoingContributionStatusService {
+
+  private final OngoingContributionStatusRepository ongoingContributionStatusRepository;
+  @Override
+  public void updateOngoingContribution(OngoingContributionStatus ongoingContributionStatus,
+                                        String errorMsg, ContributionStatus status) {
+    ongoingContributionStatus.setError(errorMsg);
+    updateStatusAndSaveOngoingJob(ongoingContributionStatus, status);
+  }
+
+  @Override
+  public void updateOngoingContribution(OngoingContributionStatus ongoingContributionStatus,
+                                        ContributionStatus status) {
+    updateStatusAndSaveOngoingJob(ongoingContributionStatus, status);
+  }
+
+  private void updateStatusAndSaveOngoingJob(OngoingContributionStatus ongoingContributionStatus,
+                                             ContributionStatus status) {
+    ongoingContributionStatus.setStatus(status);
+    ongoingContributionStatus.setRetryAttempts(status.equals(RETRY) ?
+      ongoingContributionStatus.getRetryAttempts() + 1 : ongoingContributionStatus.getRetryAttempts());
```

Comment:
Minor but do you need to get the retry attempts here? Won't it already have the right value. Maybe ternary should be replaced by a `if`.

Reply:
The purpose of getting the retry attempt is to increase the existing retry count by 1 whenever we get the status RETRY, we need to make sure the retry count is incremented. Retry attempts value indicates no of attempts in which we are trying to process a particular record.